src/views/Header/HeaderStyleInterpolator.js

```diff
@@ -17,6 +41,37 @@ const crossFadeInterpolation = (first, index, last) => ({
  * +-------------+-------------+-------------+
  */
 
+function isGoingBack(scenes) {
+  const lastSceneIndexInScenes = scenes.length - 1;
+  return !scenes[lastSceneIndexInScenes].isActive;
+}
+
+function forLayout(props) {
+  const { layout, position, scene, scenes } = props;
+  const isBack = isGoingBack(scenes);
+
+  const interpolate = getSceneIndicesForInterpolationInputRange(props);
+  if (!interpolate) return {};
+
+  const { first, last } = interpolate;
+  const index = scene.index;
+
+  const width = layout.initWidth;
+  const rtlMult = I18nManager.isRTL ? -1 : 1;
+  const translateX = position.interpolate({
+    inputRange: [first, index, last],
+    outputRange: [
+      rtlMult * (hasHeader(scenes[first]) ? 0 : width),
+      rtlMult * (hasHeader(scenes[index]) ? 0 : isBack ? width : -width),
+      rtlMult * (hasHeader(scenes[last]) ? 0 : -width),
```

Comment:
Was this tested with RTL? This constraint is can be easy to forget about.. maybe we should add some info to the contributors guide to help people test edge cases.

Reply:
I did test with RTL, though unfortunately I think the swipe to go back gesture is broken in RTL mode.